GeometryCore/src/Polygon.cpp

```diff
@@ -58,13 +70,55 @@ namespace geom_utils
     // Calculate and return convex hull for the polygon
     Polygon Polygon::convexHull() const 
     { 
-        return {}; 
+        //if (points.size() < 3) return; //can't bild a polygon from 2 points or less //idk how to do it
+        Polygon convexHull;
+        int pos_of_the_most_right_point = 0;
+
+        for (int i = 1; i < points.size(); i++) if (points[i].x < points[pos_of_the_most_right_point].x) pos_of_the_most_right_point = i; //idk how to do it with stl
+
+        int pos_of_cur_point = pos_of_the_most_right_point;
+        do
+        {
+            convexHull.points.push_back(points[pos_of_cur_point]);
+            int pos_of_next_point = (pos_of_cur_point + 1) % points.size();
+
+            for (int i = 0; i < points.size(); i++)
+            {   //find the most counter-clockwise point
+                Polygon tmp_poly({ points[pos_of_cur_point], points[i], points[pos_of_next_point] });
+                if (tmp_poly.orientation())
+                    pos_of_next_point = i;
+            }
+            pos_of_cur_point = pos_of_next_point;
+
+        } while (pos_of_cur_point != pos_of_the_most_right_point);
+
+        return convexHull;
     }
 
     // Return true if polygon contour is convex, false if non-convex
     bool Polygon::isConvexHull() const
-    { 
-        return {}; 
+    {
+        if (points.size() < 3) return false;
+        int convex_flag = 0;
+        bool isConvex = true;
+
+        #pragma omp parallel for
+        for (int i = 0; i < points.size(); i++)
+        {
+            FPoint2D v1 = points[i];
+            FPoint2D v2 = points[(i + 1) % points.size()];
+            FPoint2D v3 = points[(i + 2) % points.size()];
+            float res = (v2.x - v1.x) * (v3.y - v2.y) - (v2.y - v1.y) * (v3.x - v2.x);
+
+            #pragma omp critical
+            {
+                if (res < 0) convex_flag |= 1;
```

Comment:
This section and final check look somewhat overloaded. From my view, `Polygon::isConvexHull()` could be simplified.
Please consider the following pseudocode. 
```
return true if:
    orientation of polygon is counter clockwise and all cross products are non-negative 
        or
    orientation of polygon is clockwise and all cross products are negative
otherwise return false
```

Reply:
ok, i will try to do it